spec/acceptance/suites/default/00_default_spec.rb

```diff
@@ -28,27 +28,30 @@
       # the lower case format is needed.
       let(:results_base_dn) { fact_on(server, 'domain').split('.').map{ |d| "dc=#{d}" }.join(',') }
 
-      let(:server_hieradata)      { File.read(File.expand_path('templates/server_hieradata.yaml.erb', File.dirname(__FILE__))) }
-      let(:server_hieradata_tls)  { File.read(File.expand_path('templates/server_hieradata_tls.yaml.erb', File.dirname(__FILE__))) }
       let(:add_testuser)          { File.read(File.expand_path('templates/add_testuser.ldif.erb', File.dirname(__FILE__))) }
       let(:add_testuser_to_admin) { File.read(File.expand_path('templates/add_testuser_to_admin.ldif.erb', File.dirname(__FILE__))) }
 
       context 'default parameters (no pki)' do
-        it 'should configure server with tls disabled and with no errors' do
-          echo_on(server, base_dn)
+        let(:server_hieradata)      { ERB.new(File.read(File.expand_path('templates/server_hieradata.yaml.erb', File.dirname(__FILE__)))).result(binding) }
 
-          on(server, 'mkdir -p /usr/local/sbin/simp')
-          set_hieradata_on(server, ERB.new(server_hieradata).result(binding))
-          apply_manifest_on(server, server_manifest, :catch_failures => true)
-          apply_manifest_on(server, server_manifest, :acceptable_exit_codes => [0,2])
-          apply_manifest_on(server, server_manifest, :acceptable_exit_codes => [0,2])
+        context 'preparing for run' do
+          it 'should configure server with tls disabled and with no errors' do
+            echo_on(server, base_dn)
 
-          # reboot to apply auditd changes
-          # shell( 'shutdown -r now', { :expect_connection_failure => true } )
-        end
+            on(server, 'mkdir -p /usr/local/sbin/simp')
+
+            set_hieradata_on(server, server_hieradata)
+            apply_manifest_on(server, server_manifest, :catch_failures => true)
+            #apply_manifest_on(server, server_manifest, :acceptable_exit_codes => [0,2])
+            #apply_manifest_on(server, server_manifest, :acceptable_exit_codes => [0,2])
+
+            # reboot to apply auditd changes
+            # shell( 'shutdown -r now', { :expect_connection_failure => true } )
+          end
 
-        it 'should be idempotent' do
-          apply_manifest(server_manifest, {:catch_changes => true})
+          it 'should be idempotent' do
+            apply_manifest(server_manifest, {:catch_changes => true})
```

Comment:
Your acceptance tests are failing on this because the service.pp creates a file in bootstrap_ldap without setting the selinux context.  When it goes through the second time server.pp in the file resource for /etc/openldap will then correct the setting so it fails idempotency.  I threw an  a selinux_ignore_defaults in the file resource and the tests all passed but I don't think that is how we ant to do it.  I don't think shoving chcon -u system_u -t etc_t in the bootstrap command will work because if you have selinux disabled it would error out I think.  You can put a test in there.

Reply:
@jeannegreulich This is a bit strange. I didn't have any acceptance test issues, but I'm investigating it further to make sure whatever we do sticks.

Reply:
Found the issue and have a workaround that's only mildly horrible.